src/main/scala/com/datastax/gatling/plugin/request/CqlRequestAction.scala

```diff
@@ -61,48 +62,53 @@ class CqlRequestAction(val name: String,
     })
   }
 
-  def sendQuery(session: Session): Unit = {
-    val enableCO = Boolean.getBoolean("gatling.dse.plugin.measure_service_time")
-    val responseTimeBuilder: ResponseTimeBuilder = if (enableCO) {
-      // The throughput checker is useless in CO affected scenarios since throughput is not known in advance
-      COAffectedResponseTime.startingNow(gatlingTimingSource)
-    } else {
-      ThroughputVerifier.checkForGatlingOverloading(session, gatlingTimingSource)
-      GatlingResponseTime.startedByGatling(session, gatlingTimingSource)
-    }
-    val stmt = safely()(dseAttributes.statement.buildFromSession(session))
+  private def buildStatement(builder:B):T = {
 
-    stmt.onFailure(err => {
-      handleFailure(session, responseTimeBuilder, err)
-    })
+    // global options
+    dseAttributes.cl.foreach(builder.setConsistencyLevel)
+    dseAttributes.idempotent.foreach(builder.setIdempotence(_))
+    dseAttributes.node.foreach(builder.setNode)
 
-    stmt.onSuccess({ stmt =>
-      // global options
-      dseAttributes.cl.map(stmt.setConsistencyLevel)
-      dseAttributes.userOrRole.map(stmt.executingAs)
-      dseAttributes.readTimeout.map(stmt.setReadTimeoutMillis)
-      dseAttributes.idempotent.map(stmt.setIdempotent)
-      dseAttributes.defaultTimestamp.map(stmt.setDefaultTimestamp)
-
-      // CQL Only Options
-      dseAttributes.outGoingPayload.map(x => stmt.setOutgoingPayload(x.asJava))
-      dseAttributes.serialCl.map(stmt.setSerialConsistencyLevel)
-      dseAttributes.retryPolicy.map(stmt.setRetryPolicy)
-      dseAttributes.fetchSize.map(stmt.setFetchSize)
-      dseAttributes.pagingState.map(stmt.setPagingState)
-      if (dseAttributes.enableTrace.isDefined && dseAttributes.enableTrace.get) {
-        stmt.enableTracing
+    // CQL Only Options
+    dseAttributes.customPayload.foreach {
+      _.foreach {
+        _ match {
+          case (k, v) => builder.addCustomPayload(k, v)
+        }
       }
+    }
+    dseAttributes.enableTrace.foreach(builder.setTracing)
+    dseAttributes.pageSize.foreach(builder.setPageSize)
+    dseAttributes.pagingState.foreach(builder.setPagingState)
+    dseAttributes.queryTimestamp.foreach(builder.setQueryTimestamp)
+    dseAttributes.routingKey.foreach(builder.setRoutingKey)
+    dseAttributes.routingKeyspace.foreach(builder.setRoutingKeyspace)
+    dseAttributes.routingToken.foreach(builder.setRoutingToken)
+    dseAttributes.serialCl.foreach(builder.setSerialConsistencyLevel)
+    dseAttributes.timeout.foreach(builder.setTimeout)
+    builder.build
+  }
 
-      val responseHandler = new CqlResponseHandler(next, session, system, statsEngine, responseTimeBuilder, stmt, dseAttributes, metricsLogger)
-      implicit val sameThreadExecutionContext: ExecutionContextExecutor = ExecutionContext.fromExecutorService(dseExecutorService)
-      FutureUtils
```

Comment:
This use (and a similar use in the graph Action impl) were the sole reasons for FutureUtils existence.  The base converters seem to be working well enough to warrant a wholesale change.